internal/io/redis/lookup.go

```diff
@@ -66,13 +57,32 @@ func (s *lookupSource) Configure(datasource string, props map[string]interface{}
 		return errors.New("redis dataType must be string or list")
 	}
 	s.c = cfg
+	return nil
+}
+
+func (s *lookupSource) Configure(datasource string, props map[string]interface{}) error {
+	if datasource != "/$$TEST_CONNECTION$$" {
+		db, err := cast.ToInt(datasource, cast.CONVERT_ALL)
+		if err != nil {
+			return fmt.Errorf("invalid datasource, it must be an integer but got %s", datasource)
+		}
+		s.db = db
+	} else {
+		s.db = 0
+	}
+	if s.db < 0 || s.db > 15 {
```

Comment:
Why this is not inside the validate func

Reply:
Due to the db is passed by datasource in Configure.